addons/payment/models/payment_transaction.py

```diff
@@ -167,11 +167,13 @@ def create(self, values_list):
 
             # Duplicate partner values.
             partner = self.env['res.partner'].browse(values['partner_id'])
+            # partner_email must be unique
+            partner_email = email_normalize_all(partner.email)[0] if partner.email else None
```

Comment:
It would be self-explanatory (no need for comment) if split in two steps:
```suggestion
            partner_emails = email_normalize_all(partner.email)
```
and
```py
'partner_email': partner_emails[0] if partner_emails else None,
```